src/analyses/mallocFresh.ml

```diff
@@ -29,6 +29,23 @@ struct
   let assign ctx lval rval =
     assign_lval (Analyses.ask_of_ctx ctx) lval ctx.local
 
+    let is_relevant_variable v =
+      (* Check if the variable is a global variable *)
+      v.vglob
+    
+    let event ctx e octx =
+        match e with
+        | Events.Invalidate {lvals} ->
```

Comment:
The `lvals` here are only from the `out` set. Could the value of a pointer variable (referencing an allocated block) in the `in` set be read and copied somewhere, e.g. to a global variable, by the assembly?